test/test_proxy_lib.cpp

```diff
@@ -41,46 +42,162 @@ TEST_F(test, proxyLib_basic) {
 }
 
 TEST_F(test, proxyLib_realloc_size0) {
-    // realloc(ptr, 0) == free (ptr)
+    // realloc(ptr, 0) == free(ptr)
     // realloc(ptr, 0) returns NULL
-    ASSERT_EQ(::realloc(::malloc(SIZE_64), 0), nullptr);
+    ASSERT_EQ(::realloc(::malloc(SIZE_EQ), 0), nullptr);
 }
 
-TEST_F(test, proxyLib_malloc_usable_size) {
-
-    void *ptr = ::malloc(SIZE_64);
-    ASSERT_NE(ptr, nullptr);
-    if (ptr == nullptr) {
-        // Fix for the following CodeQL's warning on Windows:
-        // 'ptr' could be '0': this does not adhere to the specification for the function '_msize'.
-        return;
-    }
+// The proxyLib_size_threshold_* tests test the size threshold of the proxy library.
+// The size threshold is set to SIZE_THRESHOLD bytes in this test, so all allocations of:
+// 1) size <  SIZE_THRESHOLD go through the default system allocator
+//    (umfPoolByPtr(ptr_size < SIZE_THRESHOLD) == nullptr)
+// 2) size >= SIZE_THRESHOLD go through the proxy library allocator
+//    (umfPoolByPtr(ptr_size >= SIZE_THRESHOLD) != nullptr)
 
+TEST_F(test, proxyLib_size_threshold_aligned_alloc) {
 #ifdef _WIN32
-    size_t size = _msize(ptr);
-#elif __APPLE__
-    size_t size = ::malloc_size(ptr);
+    void *ptr_LT = _aligned_malloc(SIZE_LT, ALIGN_1024);
+    void *ptr_EQ = _aligned_malloc(SIZE_EQ, ALIGN_1024);
 #else
-    size_t size = ::malloc_usable_size(ptr);
+    void *ptr_LT = ::aligned_alloc(ALIGN_1024, SIZE_LT);
+    void *ptr_EQ = ::aligned_alloc(ALIGN_1024, SIZE_EQ);
 #endif
 
-    ASSERT_EQ((int)(size == 0 || size >= SIZE_64), 1);
+    ASSERT_NE(ptr_LT, nullptr);
+    ASSERT_NE(ptr_EQ, nullptr);
 
-    ::free(ptr);
-}
+    // verify alignment
+    ASSERT_EQ((int)(IS_ALIGNED((uintptr_t)ptr_LT, ALIGN_1024)), 1);
+    ASSERT_EQ((int)(IS_ALIGNED((uintptr_t)ptr_EQ, ALIGN_1024)), 1);
+
+#ifndef _WIN32 /* the size threshold works only on Linux for now */
+    // umfPoolByPtr(ptr_size_LT) == nullptr
+    ASSERT_EQ(umfPoolByPtr(ptr_LT), nullptr);
+#endif
+    // umfPoolByPtr(ptr_size_EQ) != nullptr
+    ASSERT_NE(umfPoolByPtr(ptr_EQ), nullptr);
 
-TEST_F(test, proxyLib_aligned_alloc) {
 #ifdef _WIN32
-    void *ptr = _aligned_malloc(SIZE_64, ALIGN_1024);
+    _aligned_free(ptr_LT);
+    _aligned_free(ptr_EQ);
 #else
-    void *ptr = ::aligned_alloc(ALIGN_1024, SIZE_64);
+    ::free(ptr_LT);
+    ::free(ptr_EQ);
+#endif
+}
+
+TEST_F(test, proxyLib_size_threshold_malloc) {
```

Comment:
we could have a test with multiple allocs, not just 2

Reply:
What would it change?

Reply:
we would have more tests 😄

Reply:
Only 2 are needed in tests of the size threshold: one for the default system allocator (size < threshold) and another one for the proxy library pool UMF pool allocator (size >= threshold). There is no need for an extra one.